wai-middleware-prometheus/src/Network/Wai/Middleware/Prometheus.hs

```diff
@@ -158,26 +161,32 @@ observeSeconds handler method status start end = do
 
 -- | Expose Prometheus metrics and instrument an application with some basic
 -- metrics (e.g. request latency).
-prometheus :: PrometheusSettings -> Wai.Middleware
-prometheus PrometheusSettings{..} app req respond =
+prometheus :: Bool -> Int -> PrometheusSettings -> Wai.Middleware
+prometheus enableCompression compressionLevel PrometheusSettings{..} app req respond =
```

Comment:
Ideally we shouldn't send compressionLevel when enableCompression is false. Actually i think it's better if we can put these values in PrometheusSettings itself.

Reply:
Sure, but I think we are now mostly using metricsApp for running the prometheus server rather than this function. For metricsApp, from where should we use compressionLevel? Plus, if we add in PrometheusSettings, should we hardcode the compressionLevel value?

Reply:
Now we are using metricsApp function rather than prometheus for metrics